ruby/inc/leatherman/ruby/api.hpp

```diff
@@ -0,0 +1,669 @@
+         */
+        bool is_nil(VALUE value) const;
+
+        /**
+         * Determines if the given value is true.
+         * @param value The value to check.
+         * @return Returns true if the given value is true or false if it is not.
+         */
+        bool is_true(VALUE value) const;
+
+        /**
+         * Determines if the given value is false.
+         * @param value The value to check.
+         * @return Returns true if the given value is false or false if it is not.
+         */
+        bool is_false(VALUE value) const;
+
+        /**
+         * Determines if the given value is a hash.
+         * @param value The value to check.
+         * @return Returns true if the given value is a hash or false if it is not.
+         */
+        bool is_hash(VALUE value) const;
+
+        /**
+         * Determines if the given value is an array.
+         * @param value The value to check.
+         * @return Returns true if the given value is an array or false if it is not.
+         */
+        bool is_array(VALUE value) const;
+
+        /**
+         * Determines if the given value is a string.
+         * @param value The value to check.
+         * @return Returns true if the given value is a string or false if it is not.
+         */
+        bool is_string(VALUE value) const;
+
+        /**
+         * Determines if the given value is a symbol.
+         * @param value The value to check.
+         * @return Returns true if the given value is a symbol or false if it is not.
+         */
+        bool is_symbol(VALUE value) const;
+
+        /**
+         * Determines if the given value is a fixed number (Fixnum).
+         * @param value The value to check.
+         * @return Returns true if the given value is a fixed number (Fixnum) or false if it is not.
+         */
+        bool is_fixednum(VALUE value) const;
+
+        /**
+         * Determines if the given value is a float.
+         * @param value The value to check.
+         * @return Returns true if the given value is a float or false if it is not.
+         */
+        bool is_float(VALUE value) const;
+
+        /**
+         * Gets the VALUE for nil.
+         * @return Returns the VALUE for nil.
+         */
+        VALUE nil_value() const;
+
+        /**
+         * Gets the VALUE for true.
+         * @return Returns the VALUE for true.
+         */
+        VALUE true_value() const;
+
+        /**
+         * Gets the VALUE for false.
+         * @return Returns the VALUE for false.
+         */
+        VALUE false_value() const;
+
+        /**
+         * Looks up a constant based on the given names. The individual entries correspond to
+         * namespaces, as in {"A", "B", "C"} => A::B::C.
+         * @param names The names to lookup.
+         * @return Returns the value or raises a NameError.
+         */
+        VALUE lookup(std::initializer_list<std::string> const& names) const;
```

Comment:
No, that'd be `ruby.rb_hash_lookup(ruby.lookup({"RbConfig", "CONFIG"}), ruby.rb_str_new_cstr("host_os"));` (assuming we imported `rb_str_new_cstr`).